util.go

```diff
@@ -28,12 +28,12 @@ func newSeed() int64 {
 }
 
 // randomTimeout returns a value that is between the minVal and 2x minVal.
-func randomTimeout(minVal time.Duration) <-chan time.Time {
+func randomTimeout(minVal time.Duration) timerWrapper {
 	if minVal == 0 {
-		return nil
+		return newTimer(0)
 	}
 	extra := (time.Duration(rand.Int63()) % minVal)
-	return time.After(minVal + extra)
+	return newTimer(minVal + extra)
```

Comment:
I think we can make this function work without the `timerWrapper` by returning a `noopTimer` implementation instead of `nil`. The `noopTimer` can have a method `Stop`  that does nothing, and a field `C` that is a nil channel. Then we can return the `noopTimer` or  a `*time.Timer`

Reply:
I love the idea of `noopTimer`. I would fix that.

Reply:
Thinking about this more, I'm not sure how to return this from `randomTimeout`. We could use an interface, but interfaces don't allow fields.   I'm not sure if there is any way to make this work.

Reply:
It's possible to use interface and follow SOLID principle here. But the trade-off might look like:
```go
t.C -> t.C() // return channel through func 
```
If it looks okay to you, then I think interface is a good choice rather than concrete type